### ISSUE TITLE
revise classes documentation so that they all conform to a certain starndard

### ISSUE THREAD

**Issue by fnrizzi** [Pressio/rom-tools-and-workflows#24]
out of curiosity i tried to ask chatgpt to write the documentation for the `AbstractSnapshotData`

<img width="817" alt="image" src="https://github.com/Pressio/rom-tools-and-workflows/assets/18708772/87af2fdb-c6a6-4654-9e0d-8d4c54daf50e">

and I got this response: 

```py
class AbstractSnapshotData(abc.ABC):
    """
    An abstract base class for representing snapshot data.

    This class defines the common interface for classes that store and provide access to snapshot data
    as part of a simulation or data processing system. Implementations of this class are expected to
    define the initialization method and various methods for accessing and manipulating the data.

    Attributes:
        var_names (list): A list of variable names associated with the snapshot data.

    Methods:
        __init__(self, **kwargs):
            Initializes an instance of the AbstractSnapshotData class. Subclasses should implement
            this method to set up the necessary data structures or connections to data sources.

        getSnapshotsAsListOfArrays(self) -> Iterable[np.ndarray]:
            Retrieve the snapshots as an iterable of NumPy arrays. Each array represents a single snapshot.

        getMeshGids(self):
            Retrieve the mesh global identifiers associated with the snapshots.

        getSnapshotsAsArray(self) -> np.ndarray:
            Retrieve the snapshots as a single NumPy array by converting the list of snapshots into an array.

        getVariableNames(self) -> list:
            Retrieve the list of variable names associated with the snapshot data.

        getNumVars(self) -> int:
            Get the number of variables (dimensions) in the snapshot data.

    Note:
    - Subclasses should implement the abstract methods `__init__()`, `getSnapshotsAsListOfArrays()`,
      and `getMeshGids()`.
    - The attribute `var_names` should be set in the subclass constructor to define the variable names.
    - The `getSnapshotsAsArray()` and `getNumVars()` methods provide convenient access to the data.
    """

    @abc.abstractmethod
    def __init__(self, **kwargs):
        pass

    @abc.abstractmethod
    def getSnapshotsAsListOfArrays(self) -> Iterable[np.ndarray]:
        pass

    @abc.abstractmethod
    def getMeshGids(self):
        pass

    def getSnapshotsAsArray(self) -> np.ndarray:
        snapshot_array = listOfSnapshotsToArray(self.getSnapshotsAsListOfArrays())
        return snapshot_array

    def getVariableNames(self) -> list:
        return self.var_names

    def getNumVars(self) -> int:
        return len(self.getVariableNames())
```

then i asked to do

<img width="696" alt="image" src="https://github.com/Pressio/rom-tools-and-workflows/assets/18708772/ae2809bb-97b2-46bb-b987-6fb8c9ee4fdc">

and i got this:

```py
class AbstractSnapshotData(abc.ABC):
    """
    An abstract base class for representing snapshot data.

    This class defines the common interface for classes that store and provide access to snapshot data
    as part of a simulation or data processing system. Implementations of this class are expected to
    define the initialization method and various methods for accessing and manipulating the data.

    Attributes:
        var_names (list): A list of variable names associated with the snapshot data.

    Methods:
    """

    @abc.abstractmethod
    def __init__(self, **kwargs):
        """
        Initializes an instance of the AbstractSnapshotData class. Subclasses should implement
        this method to set up the necessary data structures or connections to data sources.

        Args:
            **kwargs: Additional keyword arguments that subclasses may accept for configuration.

        Note:
        Subclasses must call this base class constructor and set the `var_names` attribute to
        define the variable names associated with the snapshot data.
        """
        pass

    @abc.abstractmethod
    def getSnapshotsAsListOfArrays(self) -> Iterable[np.ndarray]:
        """
        Retrieve the snapshots as an iterable of NumPy arrays. Each array represents a single snapshot.

        Returns:
            Iterable[np.ndarray]: An iterable of NumPy arrays representing the snapshots.

        Note:
        Subclasses must implement this method to provide access to the actual snapshot data.
        """
        pass

    @abc.abstractmethod
    def getMeshGids(self):
        """
        Retrieve the mesh global identifiers associated with the snapshots.

        Returns:
            None or specific data type: The mesh global identifiers, or None if not applicable.

        Note:
        Subclasses must implement this method to provide access to mesh global identifiers if relevant.
        """
        pass

    def getSnapshotsAsArray(self) -> np.ndarray:
        """
        Retrieve the snapshots as a single NumPy array by converting the list of snapshots into an array.

        Returns:
            np.ndarray: A NumPy array containing all the snapshots.

        Note:
        This method provides a convenient way to access the snapshot data as a single array.
        Subclasses can use the `getSnapshotsAsListOfArrays()` method to implement this.
        """
        snapshot_array = listOfSnapshotsToArray(self.getSnapshotsAsListOfArrays())
        return snapshot_array

    def getVariableNames(self) -> list:
        """
        Retrieve the list of variable names associated with the snapshot data.

        Returns:
            list: A list of variable names.

        Note:
        Subclasses should ensure that this list is properly defined and set in the constructor.
        """
        return self.var_names

    def getNumVars(self) -> int:
        """
        Get the number of variables (dimensions) in the snapshot data.

        Returns:
            int: The number of variables.

        Note:
        Subclasses should make sure that this method returns the correct number of variables
        associated with the snapshot data.
        """
        return len(self.getVariableNames())
```

**Comment by fnrizzi** [comment]
it would be good to spend a bit of time doing this for all the PUBLIC classes in the code 
BUT rather than blindly copying and pasting the result, we could use this a base to build on and also incorporate some of the documentation that Eric already wrote manually into it. 

However, the advantage of doing this is that it saves time and and make things consistent across

**Comment by cwschilly** [comment]
Closed with PRs #33, #34, #35, #36, #37, #38, and #39
